sdk/search/Azure.Search.Documents/apis.txt

```diff
@@ -31,34 +31,34 @@ namespace Azure.Search.Documents
 
         Response<AutocompleteResults> Autocomplete(string, string, AutocompleteOptions, CancellationToken) { }
         Task<Response<AutocompleteResults>> AutocompleteAsync(string, string, AutocompleteOptions, CancellationToken) { }
-        
+
         Response<IndexDocumentsResult> IndexDocuments(IndexDocumentsBatch<SearchDocument>, IndexDocumentsOptions, CancellationToken) { }
         Task<Response<IndexDocumentsResult>> IndexDocumentsAsync(IndexDocumentsBatch<SearchDocument>, IndexDocumentsOptions, CancellationToken) { }
         Response<IndexDocumentsResult> IndexDocuments<T>(IndexDocumentsBatch<T>, IndexDocumentsOptions, CancellationToken) { }
         Task<Response<IndexDocumentsResult>> IndexDocumentsAsync<T>(IndexDocumentsBatch<T>, IndexDocumentsOptions, CancellationToken) { }
     }
 
-    public class FieldBuilder { }
-    // TODO: FieldBuilder's attributes like [Key], [Searchable], etc.
+    public class SearchIndexBuilder { }
```

Comment:
I like the idea, but think this might be too presumptuous. Given a model, we can easily (at least in .NET and Java) construct fields, but other properties of indexes perhaps not. We can strive for it.

Reply:
I just think the name `FieldBuilder` is less obvious.  We can noodle on this for .NET and Java.

Reply:
The name here made me feel like it uses to build `SearchIndexClient`

Reply:
We don't use the suffix `Builder` very often in .NET so it doesn't have that connotation here.  We could pick another type name for Java if this doesn't fit.

Reply:
There are lots of details in indexes that aren't fields, and don't make sense to be tied to a model type (scoring profiles, CORS settings, custom analyzers, etc.), so I'd suggest sticking with `FieldBuilder` or `SearchFieldBuilder` if we've lost all faith in namespaces. :)

Reply:
We discussed in a meeting and will call this `SearchFieldBuilder` to group this with `SearchField` and close to `SearchIndex` in IntelliSense and IS-like implementations in other IDEs.